classes/local/updatemetadata_form.php

```diff
@@ -0,0 +1,149 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Upload video form.
+ *
+ * @package    block_opencast
+ * @copyright  2019 Farbod Zamani, ELAN e.V.
+ * @author     Farbod Zamani
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+namespace block_opencast\local;
+
+defined('MOODLE_INTERNAL') || die();
+
+global $CFG;
+
+require_once($CFG->dirroot . '/lib/formslib.php');
+
+class updatemetadata_form extends \moodleform {
+
+    public function definition() {
+
+        $mform = $this->_form;
+
+        $metadata = $this->_customdata['metadata'];
+        
+        unset($metadata[array_search('isPartOf', array_column($metadata,'id'))]);
```

Comment:
Why is this form working so differently than the addvideo_form in terms of metadata handling? Most of the fields here have no $mform->setType call.

Reply:
Updatemetadata relies completely on Opencast server metadata and gets its values from an API call. However, setType is missing and it is fixed!

Reply:
However, even if for instance the 'language'-field is returned by the metadata API, the administrator should have the possibilities to define, if it is actually returned to the user and editable in this view. So why don't we simply render the same (a similar) form than for adding a video and fill it with default values based on the return of the metadata API?